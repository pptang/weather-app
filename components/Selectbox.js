// @flow
import * as React from 'react';

type Props = {
  onSelect: Function,
};

function Selectbox(props: Props) {
  return (
    <select
      className="selectbox"
      onChange={event => {
        props.onSelect(event.target.value);
      }}
    >
      <option value="tokyo">Tokyo</option>
      <option value="new york">New York</option>
      <option value="taipei">Taipei</option>
      <option value="sydney">Sydney</option>
      <option value="seoul">Seoul</option>
      <option value="shanghai">Shanghai</option>
      <style jsx>
        {`
          .selectbox {
            height: 30px;
            margin-left: 10px;
            background-color: #ffffff;
          }
        `}
      </style>
    </select>
  );
}

export default Selectbox;
