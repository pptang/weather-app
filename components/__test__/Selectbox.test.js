import renderer from 'react-test-renderer';
import React from 'react';

import Selectbox from '../Selectbox';

describe('Pages With Snapshot Testing', () => {
  it('snapshot Selectbox Component', () => {
    const component = renderer.create(<Selectbox />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
