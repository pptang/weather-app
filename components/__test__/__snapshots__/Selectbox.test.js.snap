// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot Selectbox Component 1`] = `
<select
  className="jsx-2805902857 selectbox"
  onChange={[Function]}
>
  <option
    className="jsx-2805902857"
    value="tokyo"
  >
    Tokyo
  </option>
  <option
    className="jsx-2805902857"
    value="new york"
  >
    New York
  </option>
  <option
    className="jsx-2805902857"
    value="taipei"
  >
    Taipei
  </option>
  <option
    className="jsx-2805902857"
    value="sydney"
  >
    Sydney
  </option>
  <option
    className="jsx-2805902857"
    value="seoul"
  >
    Seoul
  </option>
  <option
    className="jsx-2805902857"
    value="shanghai"
  >
    Shanghai
  </option>
</select>
`;
